### PR TITLE
[PoC] Reorganise docs

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

- Reorganizes docs to have one page per metrics
- Updates index page to match pytorch lightning
- Adds a metrics page where metrics can be filtrered by tag

Follow-up work to do:

- Add descriptions for each metric
- Update metric images to be more relevant
- Add more tags to each metric (currently just one tag per metric)

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Review by stancld — APPROVED** [review]
The result looks really lovely!

File: docs/source/index.rst [hunk 1]
```diff
@@ -2,14 +2,125 @@
...
 
-.. include:: pages/brief_intro.rst
+.. raw:: html
 
+   <div class="row" style='font-size: 14px'>
+      <div class='col-md-12'>
 
-More reading
-============
+TorchMetrics is a collection of 80+ PyTorch metrics implementations and an easy-to-use API to create custom metrics. It offers:
```
Comment (tchaton) [inline docs/source/index.rst:13]: Let's update the readme too.

**Review by tchaton — APPROVED** [review]
Neat !

**Review by justusschock — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: docs/source/index.rst (hunk 1 above)
Comment (Borda) [reply docs/source/index.rst:13]: what do you mean?

**Pull request merged**